core/authority_discovery/publisher/address_publisher.cpp

```diff
@@ -117,6 +117,26 @@ namespace kagome::authority_discovery {
       return outcome::success();
     }
 
+    OUTCOME_TRY(
+        raw,
+        audiEncode(ed_crypto_provider_,
+                   sr_crypto_provider_,
+                   *libp2p_key_,
+                   *libp2p_key_pb_,
+                   peer_info,
+                   *audi_key,
+                   std::chrono::system_clock::now().time_since_epoch()));
+    return kademlia_->putValue(std::move(raw.first), std::move(raw.second));
+  }
+
+  outcome::result<std::pair<Buffer, Buffer>> audiEncode(
```

Comment:
Consider creating KeyPairs struct and CryptoProviders sturcts to reduce arguments number at most to 5